lib/core/test/unit/Cardano/Wallet/Primitive/Migration/SelectionSpec.hs

```diff
@@ -0,0 +1,1036 @@
+  where
+    ArgsForCreate
+        { mockConstraints
+        , mockInputs
+        , mockRewardWithdrawal
+        } = args
+    constraints = unMockTxConstraints mockConstraints
+    result = create constraints
+        (RewardWithdrawal mockRewardWithdrawal)
+        (mockInputs)
+
+resultIsSelection :: MockSelectionResult -> Bool
+resultIsSelection = isRight
+
+resultHasZeroFeeExcess :: MockSelectionResult -> Bool
+resultHasZeroFeeExcess = matchRight $ \selection ->
+    feeExcess selection == Coin 0
+
+resultHasInsufficientAda :: MockSelectionResult -> Bool
+resultHasInsufficientAda = matchLeft $ \case
+    SelectionAdaInsufficient -> True
+    _ -> False
+
+resultIsFull :: MockSelectionResult -> Bool
+resultIsFull = matchLeft $ \case
+    SelectionFull _ -> True
+    _ -> False
+
+--------------------------------------------------------------------------------
+-- Extending a selection
+--------------------------------------------------------------------------------
+
+data ArgsForExtend = ArgsForExtend
+    { mockConstraints :: MockTxConstraints
+    , mockSelection :: MockSelection
+    , mockInput :: (MockInputId, TokenBundle)
+    } deriving (Eq, Show)
+
+instance Arbitrary ArgsForExtend where
+    arbitrary = genArgsForExtend
+
+genArgsForExtend :: Gen ArgsForExtend
+genArgsForExtend = genInner `suchThatMap` id
+  where
+    genInner = do
+        ArgsForCreate {mockConstraints, mockInputs, mockRewardWithdrawal}
+            <- genArgsForCreate
+        let constraints = unMockTxConstraints mockConstraints
+        mockInput <- (,)
+            <$> genMockInputId
+            <*> oneof
+                [ genTokenBundleMixed mockConstraints
+                  -- In order to increase coverage of error conditions,
+                  -- deliberately include some large bundles whose ada
+                  -- quantities are below the minimum:
+                , TokenBundle (Coin 0) . F.fold <$>
+                    replicateM 4 (genTokenMap mockConstraints)
+                ]
+        let maybeSelection = create constraints
+                (RewardWithdrawal mockRewardWithdrawal) mockInputs
+        pure $ case maybeSelection of
+            Left _ -> Nothing
+            Right mockSelection -> Just ArgsForExtend
+                { mockConstraints
+                , mockSelection
+                , mockInput
+                }
+
+prop_extend :: ArgsForExtend -> Property
+prop_extend args =
+    checkCoverage $
+    cover 40 (resultIsSelection result)
+        "Success" $
+    cover 10 (resultHasZeroFeeExcess result)
+        "Success with zero fee excess" $
+    cover 0.1 (resultHasInsufficientAda result)
+        "Failure due to insufficient ada" $
+    cover 0.1 (resultIsFull result)
+        "Failure due to oversized selection" $
+    case result of
+        Left SelectionAdaInsufficient ->
+            property True
+        Left (SelectionFull e) ->
+            property (selectionSizeMaximum e < selectionSizeRequired e)
```

Comment:
Suggestion: also call `Selection.create` here to verify that this function will also produce the same error.